File: 107 Binary Tree Level Order Traversal II/107.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> res = new LinkedList<List<Integer>>();
        if (root == null) {
            return res;
        }
        helper(res, root, 0);
        return res;
    }
    
    private void helper(List<List<Integer>> res, TreeNode root, int level) {
        if (root == null) {
            return;
        }
        if (level >= res.size()) {
            // 
            res.add(0, new ArrayList<Integer>());
        }
        helper(res, root.left, level + 1);
        helper(res, root.right, level + 1);
        res.get(res.size() - 1 - level).add(root.val);
    }
}

File: 107 Binary Tree Level Order Traversal II/107.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrderBottom = function(root) {
    let res = []
    if (!root) {
        return res;
    }
    helper(res, root, 0)
    return res;
};

var helper = function(res, root, level) {
    if (!root) {
        return;
    }
    
    if (level >= res.length) {
        res.unshift([])
    }
    
    helper(res, root.left, level + 1)
    helper(res, root.right, level + 1)
    res[res.length - 1 - level].push(root.val)
}